File: src/704BinarySearch/solution.cpp
class Solution {
public:
    int search(vector<int> &nums, int target) {
        int lo = 0, hi = nums.size() - 1;
        while (lo <= hi) {
            int mid = lo + ((hi - lo) >> 1);
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] > target) {
                hi = mid - 1;
            } else {
                lo = mid + 1;
            }
        }
        return -1;
    }
};

File: src/704BinarySearch/solution.kt
package `704BinarySearch`

class Solution {
    fun search(nums: IntArray, target: Int): Int {
        val index = -1
        var begin = 0
        var end = nums.size - 1
        var middle = (end + begin) / 2
        while (begin <= end) {
            if (nums[middle] == target) return middle
            if (nums[middle] > target) {
                end = middle - 1
                middle = (end + begin) / 2
            }
            if (nums[middle] < target) {
                begin = middle + 1
                middle = (end + begin) / 2
            }
        }
        return index
    }
}